tests/js/session-tests.js

```diff
@@ -461,6 +461,25 @@ module.exports = {
         });
     },
 
+    testIncompatibleSyncedRealm() {
+        return Realm.Sync.User.register('http://localhost:9080', uuid(), 'password').then(user => {
+            return new Promise((resolve, _reject) => {
+                const config = { path: 'sync-1.x.realm', sync: { user, url: 'realm:://localhost:9080/~/sync-1.x' } };
+                try {
+                    const realm = new Realm(config);
+                }
+                catch (e) {
+                    if (e.match('/IncompatibleSyncedRealm: /')) {
+                        const backupConfig = { path: e.substring('IncompatibleSyncedRealm: '.length) };
```

Comment:
Doesn't this configuration need to specify that the Realm should be opened as read-only? If you don't do that you should get an incompatible histories exception from opening a synced Realm as a writeable non-synced Realm.